src/types/redis_bloom_chain.cc

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "redis_bloom_chain.h"
+
+namespace redis {
+
+std::string BloomChain::getBFKey(const Slice &ns_key, const BloomChainMetadata &metadata, uint16_t filters_index) {
+  std::string sub_key;
+  sub_key.append(kBloomFilterSeparator);
+  PutFixed16(&sub_key, filters_index);
+
+  std::string bf_key = InternalKey(ns_key, sub_key, metadata.version, storage_->IsSlotIdEncoded()).Encode();
+  return bf_key;
+}
+
+void BloomChain::getBFKeyList(const Slice &ns_key, const BloomChainMetadata &metadata,
+                              std::vector<std::string> &bf_key_list) {
+  for (u_int16_t i = 0; i < metadata.n_filters; ++i) {
+    std::string bf_key = getBFKey(ns_key, metadata, i);
+    bf_key_list.push_back(std::move(bf_key));
+  }
+}
+
+rocksdb::Status BloomChain::getBloomChainMetadata(const Slice &ns_key, BloomChainMetadata *metadata) {
+  return Database::GetMetadata(kRedisBloomFilter, ns_key, metadata);
+}
+
+rocksdb::Status BloomChain::createBloomChain(const Slice &ns_key, double error_rate, uint32_t capacity,
+                                             uint16_t expansion, BloomChainMetadata &sb_chain_metadata) {
+  sb_chain_metadata.n_filters = 1;
+  sb_chain_metadata.expansion = expansion;
+  sb_chain_metadata.size = 0;
+
+  sb_chain_metadata.error_rate = error_rate;
+  sb_chain_metadata.base_capacity = capacity;
+  sb_chain_metadata.bloom_bytes = BlockSplitBloomFilter::OptimalNumOfBytes(capacity, error_rate);
+
+  std::string bf_data;
+  BlockSplitBloomFilter block_split_bloom_filter;
+  block_split_bloom_filter.Init(sb_chain_metadata.bloom_bytes);
+  bf_data = block_split_bloom_filter.GetData();
```

Comment:
`GetData` returns a `const std::string&`, we need take it carefully and try our best not to copying these bytes:

```c++
 batch->Put(bf_key, block_split_bloom_filter.GetData());
```

The code above is ok in Line 70